src/NodaTime.Test/TestHelper.cs

```diff
@@ -506,6 +508,23 @@ internal static void AssertBinaryRoundtrip<T>(T value)
 #endif
         }
 
+#if !NETCORE
+        internal static void AssertBinaryDeserializationFailure<T>(Action<SerializationInfo> fillInfoAction)
+        {
+            var info = new SerializationInfo(typeof(T), new FormatterConverter());
+            fillInfoAction(info);
+            var ctor = typeof(T).GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, null,
+                new Type[] { typeof(SerializationInfo), typeof(StreamingContext) }, null);
+            Assert.NotNull(ctor);
+            var context = new StreamingContext();
+            var exception = Assert.Throws<TargetInvocationException>(() => ctor.Invoke(new object[] { info, context }));
+            Assert.That(exception.InnerException, 
+                Is.InstanceOf<OverflowException>()
```

Comment:
This seems prone to us testing for one exception but getting another by mistake. Could we specify the expected exception at the callsites?

(possibly consider this as a followup, since what we have here is strictly better than what we already had.)

Reply:
Can do, will just need to refactor a bit.